checks/raw/security_policy.go

```diff
@@ -112,3 +137,87 @@ func isSecurityPolicyFilename(name string) bool {
 		strings.EqualFold(name, "doc/security.rst") ||
 		strings.EqualFold(name, "docs/security.rst")
 }
+
+var checkSecurityPolicyFileContent fileparser.DoWhileTrueOnFileContent = func(path string, content []byte,
+	args ...interface{},
+) (bool, error) {
+	if len(args) != 1 {
+		return false, fmt.Errorf(
+			"checkSecurityPolicyFileContent requires exactly one argument: %w", errInvalidArgLength)
+	}
+	pfiles, ok := args[0].(*[]checker.File)
+	if !ok {
+		return false, fmt.Errorf(
+			"checkSecurityPolicyFileContent requires argument of type *[]checker.File: %w", errInvalidArgType)
+	}
+
+	if len(content) == 0 {
+		// perhaps there are more policy files somewhere else,
+		// keep looking (true)
+		return true, nil
+	}
+
+	// TODO: there is an assertion here that if content has length
+	//       then (*pfiles)[0] is not nil (can that be checked)
+	//       (so is this test necessary?)
+	if (*pfiles) != nil {
+		// preserve file type
+		tempType := (*pfiles)[0].Type
+		linkedContentLen, urls, emails, discvuls := countUniquePolicyHits(string(content))
+		contentMetrics := fmt.Sprintf("%d,%d,%d,%d,%d", len(content), linkedContentLen, urls, emails, discvuls)
+		(*pfiles)[0] = checker.File{
+			Path:    path,
+			Type:    tempType,
+			Offset:  checker.OffsetDefault,
+			Snippet: contentMetrics,
```

Comment:
Snippet is typically used for exact snippet of code. I suggest we instead update this structure https://github.com/ossf/scorecard/blob/main/checker/raw_result.go#L141-L144 to contain the information gathered in this file. Then in the evaluation part, we use that.

How about:
```golang

type SecurityPolicyInformationType string

const (
	// SecurityPolicyInformationTypeEmail blabla
	SecurityPolicyInformationType SecurityPolicyInformationTypeEmail = "emailAddress"
	SecurityPolicyInformationType SecurityPolicyInformationTypeLink = "httpLink"
        ...
)
type SecurityPolicyFileInformation struct {
   File File
   Information SecurityPolicyInformationType
}
type SecurityPolicyData struct {
	// Files contains a list of files.
	InformationFiles []SecurityPolicyFileInformation
}
```

This is not perfect, and I think we may want to update the `File` structure to
```golang
type FileLocation struct{
  Snippet   string   // Snippet of code
  Offset    uint     // Offset in the file of Path (line for source/text files).
  EndOffset uint     
}
type File struct {
	Path      string
	Locations []FileLocation
	Type      FileType // Type of file.
	// TODO: add hash.
}
```

but this is out of the scope for this PR. /cc @raghavkaul 

We can also update https://github.com/ossf/scorecard/blob/32d6ba27757fcaf6145b03195f84c74b311e4121/pkg/json_raw_results.go#L42-L47 once we've agreed what the format should be.

Reply:
I like this very much.. I new I has co-opted Snippet as a channel to get info from `raw` to `evaluation` code and did so (perhaps) violating intent of that struct field without having to change more to be cleaner -- this suggestion is much cleaner. So, Will do!

Reply:
The security policy check, unit-test, e2e tests have all be revised to reflect the change to `SecurityPolicyData`